### PR TITLE
Update keychain to extend interface and security posture

### PR DESCRIPTION
**Pull request by bashhack**
## Summary by CodeRabbit

* **New Features**
  * Enhanced keychain operations with improved security by handling secrets as byte slices and securely zeroing sensitive data after use.
  * Added support for metadata management of keychain entries, including storing, loading, and removing metadata.
  * Introduced new utilities for securely executing commands and handling sensitive input.
  * Added a cryptographically secure random string generator utility.

* **Bug Fixes**
  * Improved error handling and standardized error messages to avoid leaking sensitive information.

* **Tests**
  * Significantly expanded and refactored test coverage for all keychain and secure memory operations, including robust mocking of external commands and edge case handling.
  * Updated AWS authentication tests to use the new random string utility.

* **Documentation**
  * Added detailed comments and documentation on secure memory handling and limitations within Go's environment.

### CODE REVIEW

File: internal/keychain/metadata.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		return constants.AWSServicePrefix
+	}
+
+	// Handle unknown prefix - expected format is 'sesh-type-name'
+	parts := strings.SplitN(service, "-", 3)
+	// If we have at least 2 parts, return the first 2 joined by dash as the prefix
+	if len(parts) > 2 {
+		return fmt.Sprintf("%s-%s", parts[0], parts[1])
+	}
+	return service
```
[the remark this answers is not in the document]
Comment (bashhack) [reply internal/keychain/metadata.go:242]: Hmm- I think my logic might be correct as is?

 If I change from  > 2 to >= 2 wouldn't that cause me to extract a prefix even when the service only has 2 parts (e.g., "sesh-aws" would become "sesh-aws" which is redundant).

File: internal/keychain/metadata.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+				"-U",
+				"-T", execPath,
+			)
+			if err = cmd.Run(); err != nil {
+				return fmt.Errorf("failed to update metadata in keychain: %w", err)
+			}
+		} else {
+			return fmt.Errorf("failed to store metadata in keychain: %w", err)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (bashhack) [reply internal/keychain/metadata.go:217]: I don't remember if I was being defensive here or if I wrote this before I was using `-U`...

Whatever the case, I think your correct that I can remove this for now.
